Slightly rephrase transparency definition and add more examples (#53)
panrg/path-properties: draft-irtf-panrg-path-properties.md

: A service function that a path element applies to a flow, see {{RFC7665}}. Examples of abstract service functions include firewalls, Network Address Translation (NAT), and TCP optimizers. Some stateful service functions, such as NAT, need to observe the same flow in both directions, e.g., by being an element of both the path and the reverse path.

Transparency:

> : A node is transparent to a flow with respect to some (meta-)information for an action if the node performs this action independently of the given (meta-)information.

Reviewer: I found this sentence hard to parse.
How about something like:
```suggestion
: When a node performs an action on a flow, the node is transparent to the flow with respect to some (meta-)information if the node performs this action independently of the given (meta-)information.
```

(Meta-)information can for example be the existence of a protocol (header) in a packet or the content of a protocol header, payload, or both.
Actions can for example be blocking packets or reading and modifying (other protocol) headers or payloads.

> An IP router could be transparent to transport protocol headers such as TCP/UDP but not transparent to IP headers as it actively modifies them.

Reviewer: Just reading the whole thing again: Why does an IP router actively modify IP headers, whereby the text reads as if that's necessarily always the case?
Wouldn't the router just look up the destination address in its forwarding table and then forward the packet, modifying the Ethernet header for sure, but not the IP header?
Sorry, am I just missing something right now? :sweat_smile:

Reviewer: I guess the IP router reads these headers for sure, so the example wouldn't have to change much to say this.

Author: I was thinking that the IP router changes certain IP header fields (e.g., TTL, IP checksum), however maybe we can make a better example by considering the action of "forwarding on a certain interface".
What do you think?

```suggestion
An IP router could be transparent to transport protocol headers such as TCP/UDP but not transparent to IP headers since its forwarding behavior depends on the IP headers.
```

Reviewer: I like your suggestion. And you're right, "modify IP header" doesn't necessarily mean addresses... but still I think your proposed wording is clearer.

> A TCP SYN flooding firewall that drops TCP SYN packets after reaching a limit is transparent to IP but not to TCP/UDP.

Reviewer: If the firewall indeed does not look at the IP addresses at all, it would help to clarify the example, e.g.:
```suggestion
A TCP SYN flooding firewall that drops TCP SYN packets after having seen a certain number of packets with TCP SYN, regardless of IP address, is transparent to IP but not to TCP/UDP.
```

But it doesn't sound like a good idea to implement it this way, as that means you could DoS a network by SYN-flooding it. 
So perhaps this firewall should count TCP SYN packets from a particular IP address or for a combination of source and destination IP addresses, in which case it would not be transparent to either TCP or IP.

What do you think?

Author: I wanted to have an example where a node is transparent to IP but not to TCP/UDP to show that such a combination is also possible.
But I agree that the example does not make much sense in practice ^^

Maybe we could use the example of a firewall that only allows outgoing TCP connection, i.e., blocks all incoming TCP SYN packets regardless of the IP address?
That seems more reasonable.

Reviewer: Yep, sounds like this is a better example.

> And a NAT that actively modifies IP and TCP/UDP header information based on their content is not transparent to either IP or TCP/UDP headers.

Reviewer: Nit, I don't like sentences starting with "And" :)
```suggestion
Finally, a NAT that actively modifies IP and TCP/UDP header information based on their content is not transparent to either IP or TCP/UDP headers.
```

> Note that according to this definition, a node that modifies packets in accordance with the hosts, such as a transparent HTTP proxy, as defined in {{RFC2616}}, and a node listening and acting to implicit or explicit signals, see {{RFC8558}}, would not be considered transparent.

Reviewer: Nit, I don't think we need the hypothetical here, and the node is reacting, right? Or maybe "acting on"?
```suggestion
Note that according to this definition, a node that modifies packets in accordance with the hosts, such as a transparent HTTP proxy, as defined in {{RFC2616}}, and a node listening and reacting to implicit or explicit signals, see {{RFC8558}}, are not considered transparent.
```
